datasafe-business-tests/datasafe-business-tests-random-actions/src/test/java/de/adorsys/datasafe/business/impl/e2e/randomactions/framework/fixture/dto/TestFileTreeOper.java

```diff
@@ -0,0 +1,178 @@
+package de.adorsys.datasafe.business.impl.e2e.randomactions.framework.fixture.dto;
+
+import com.google.common.collect.Iterables;
+import de.adorsys.datasafe.business.impl.e2e.randomactions.framework.fixture.generator.RandomUsers;
+import lombok.Data;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Represents users' file tree (virtual tree).
+ */
+@Data
+@Slf4j
+public class TestFileTreeOper {
+
+    private final Consumer<Operation> publishTo;
+    private final TestUser testUser;
+    private final StorageType storageType;
+    private final Random random;
+    private final Map<String, ContentId> files = new HashMap<>();
+
+    public boolean isEmpty() {
+        return files.isEmpty();
+    }
+
+    /**
+     * Perform write operation with some content on virtual tree.
+     */
+    @Synchronized
```

Comment:
Just asking: why synchronized? does this not make the idea of parallel testing wrong?

Reply:
It is fixture **generation** method, not parallel test, it is not threadsafe without synchorization

Reply:
I guessed you know what you are doing.